### PR TITLE
Autocomplete with Suggestions

### PR DESCRIPTION
**Pull request by gayanDeegala**

### CODE REVIEW

File: src/validator/nodes/BaseNode.ts [hunk 1]
```diff
@@ -5,15 +5,16 @@ export type BaseNodes = { [key: string]: BaseNode };
...
-      this.data = data;
-      this.isRequired = isRequired;
-    }
-  }
\ No newline at end of file
+  public type?: DataType;
+  public description?: string;
+  public data: IData;
+  public isRequired?: boolean;
+  public possibleValues?: string[] | undefined;
```
Comment (malindakpt) [inline src/validator/nodes/BaseNode.ts:12]: possibleValues is not relevant for all types. Why it is moved to BaseNode?

Comment (gayanDeegala) [reply src/validator/nodes/BaseNode.ts]: this has changed

File: src/components/JsonEditorContainer/JsonEditorContainer.tsx [hunk 2]
```diff
@@ -77,12 +78,13 @@ export function JsonEditorContainer(props: { json: any, templates: Template[] })
...
                 if (item.text === "Insert") {
-                    item.text = "Prepend Item";
+                    item.click = undefined;
                     item.submenu = createValidInsertMenu(item.submenu, validInsertItems, existingKeys);
                 }
-                // adding samw logic to Append
-                if (item.text === "Append") {
-                    item.text = "Append Item";
+                // adding same logic to Append
+                if (node.type === "append" && item.text === "Append") {
```
Comment (malindakpt) [inline src/components/JsonEditorContainer/JsonEditorContainer.tsx:85]: Can we use `else if` here

**Review by malindakpt — APPROVED** [review]
Looks Good

File: src/components/JsonEditorContainer/JsonEditorContainer.tsx (hunk 2 above)
Comment (gayanDeegala) [reply src/components/JsonEditorContainer/JsonEditorContainer.tsx]: Yup, I'll do the change

**Pull request merged**